emscripten.py

```diff
@@ -1707,16 +1709,21 @@ def math_fix(g):
 STACKTOP = STACK_BASE + TOTAL_STACK;
 STACK_MAX = STACK_BASE;
 HEAP32[%d >> 2] = STACKTOP;
-Runtime.stackAlloc = asm['stackAlloc'];
-Runtime.stackSave = asm['stackSave'];
-Runtime.stackRestore = asm['stackRestore'];
-Runtime.establishStackSpace = asm['establishStackSpace'];
 ''' % shared.Settings.GLOBAL_BASE)
-
+  if settings['BINARYEN_ASYNC_COMPILATION']:
+    funcs_js.append("__ATINIT__.unshift(function() {\n"
```

Comment:
There is: https://github.com/jgravelle-google/emscripten/blob/1d81afeb464f9247698eff4780f0f3ef7781c164/emscripten.py#L1326

It uses `Module` instead of `asm`, and we just get done setting `Module["setTempRet0"]` to be a function that proxies a call to `Module["asm"]["setTempRet0"]`. `asm`/`Module["asm"]` hasn't been set yet for async compilation, but when thunking it there's no problems?

Trying that for wasm backend causes regressions from this PR for reasons I don't fully understand (specifically not having a `setTempRet0` on the global asm object).
... Ok digging deeper into that, this fixes library.js functions that with asmjs get included in the module, such as `saveSetjmp`. The specific part here is `asm = Module['asm']` at init, which should probably go in receiveInstance. So that would be a problem for asm2wasm if the `asm` references in library asm code didn't get sucked in to the wasm code.